homeassistant/components/automation/__init__.py

```diff
@@ -585,8 +589,12 @@ def log_cb(level, msg, **kwargs):
         variables = None
         if self._trigger_variables:
             try:
+                states = self.hass.states.get(self.entity_id)
+                this = states.as_dict() if states else {"entity_id": self.entity_id}
                 variables = self._trigger_variables.async_render(
-                    self.hass, None, limited=True
+                    self.hass,
+                    {"this": this},
+                    limited=True,
```

Comment:
This should not be inside the `if self._trigger_variables:`, we should always add the `this` variables:
```diff
-        variables = None
+        # Make sure the automation's state is in the state machine
+        self.async_write_ha_state()
+        this = self.hass.states.get(self.entity_id).as_dict()
+        variables = {"this": this}
         if self._trigger_variables:
             try:
-                states = self.hass.states.get(self.entity_id)
-                this = states.as_dict() if states else {"entity_id": self.entity_id}
                 variables = self._trigger_variables.async_render(
                     self.hass,
-                    {"this": this},
+                    variables,
                     limited=True,
                 )
```

Reply:
Changed.